api/envoy/api/v2/core/grpc_service.proto

```diff
@@ -53,11 +53,72 @@ message GrpcService {
     //
     //    streams_total, Counter, Total number of streams opened
     //    streams_closed_<gRPC status code>, Counter, Total streams closed with <gRPC status code>
-    string stat_prefix = 3 [(validate.rules).string.min_bytes = 1];
+    string stat_prefix = 2 [(validate.rules).string.min_bytes = 1];
 
     // Additional configuration for site-specific customizations of the Google
     // gRPC library.
-    google.protobuf.Struct config = 4;
+    google.protobuf.Struct config = 3;
+
+    // The name of the Google gRPC credentials factory to use. This must have been registered with
+    // Envoy. If this is empty, a default credentials factory will be used that sets up channel
+    // credentials based on other configuration parameters.
+    string credentials_factory_name = 4;
+
+    message ChannelCredentials {
+      oneof credential_specifier {
+        option (validate.required) = true;
+        SslCredentials ssl_credentials = 1;
+
+        // https://grpc.io/grpc/cpp/namespacegrpc.html#a6beb3ac70ff94bd2ebbd89b8f21d1f61
+        google.protobuf.Empty google_default = 2;
+      }
+    }
+
+    message CallCredentials {
```

Comment:
Maybe add comments explaining the per-channel vs. per-call distinction (and how they can combine).